fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2694,6 +2705,56 @@ public void addRebuildRecordCountsJob(List<CompletableFuture<Void>> work) {
         work.add(future);
     }
 
+    /**
+     * Boundaries are a list of key ranges maintained by each FDB server. This information can be useful for splitting a
+     * large task (e.g. rebuild index for a large record store) into small tasks (e.g.  rebuild index for records in
+     * certain primary key ranges) more evenly so that they can be executed in a parallel fashion efficiently. The
+     * returned boundaries are an estimate from FDB's locality API and may not represent the exact boundary locations
+     * at any database version.
+     *
+     * The boundaries are returned as a list, which is sorted and does not have duplication. The first element of the
+     * list is <code>low</code> and the last element is the <code>high</code>.
+     *
+     * @param low low endpoint of primary key range (inclusive)
+     * @param high high endpoint of primary key range (exclusive)
+     * @return the list of boundaries of primary keys
+     */
+    @Nonnull
+    public CompletableFuture<List<Tuple>> getPrimaryKeyBoundariesAsync(@Nonnull Tuple low, @Nonnull Tuple high) {
```

Comment:
I feel like this should probably be marked as `EXPERIMENTAL`. In particular, I could see us wanting to make this return a `RecordCusror<Tuple>` after giving it a try.

Should there also be a synchronous version that supplies the wait? And should the signature go into `FDBRecordStoreBase`?

Reply:
Yeah, I'm ok without the synchronous version.